lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp

```diff
@@ -117,15 +148,22 @@ struct InitializePythonRAII {
       PyImport_AppendInittab("_lldb", LLDBSwigPyInit);
     }
 
+#if (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 8) || (PY_MAJOR_VERSION > 3)
+    config.install_signal_handlers = 0;
+    Py_InitializeFromConfig(&config);
+    PyConfig_Clear(&config);
+    InitializeThreadsPrivate();
+#else
 // Python < 3.2 and Python >= 3.2 reversed the ordering requirements for
 // calling `Py_Initialize` and `PyEval_InitThreads`.  < 3.2 requires that you
 // call `PyEval_InitThreads` first, and >= 3.2 requires that you call it last.
-#if (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 2) || (PY_MAJOR_VERSION > 3)
+#if (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 2)
```

Comment:
Doesn't this now do the wrong thing for a hypothetical 4.x? I thought that's what the `|| (PY_MAJOR_VERSION > 3)` was there for.

Reply:
[#if PY_VERSION_HEX>=0x03020000](https://github.com/python/cpython/blob/dcad8fecbd5c8f1542f7a46e0c7bb7e1f27ab115/Include/patchlevel.h#L31) ?

Reply:
@DavidSpickett The hypothetical Python 4 case is covered above and goes down the ` Py_InitializeFromConfig` path. 

@labath Happy to use that but everything else in this file uses the same check I have here, so I'll simplify the defines in a separate patch/PR.

Reply:
Yes you're right I was struggling to read the nested ifdefs. This 3.2 code will be removed once the 20 branch happens so it's fine as is.